tools/payloads/ysoserial/find_ysoserial_offsets.rb

```diff
@@ -122,103 +122,134 @@ def get_payload_list
-      end
+    # Comparing diffs of various payload lengths to find length and buffer offsets
+    (PAYLOAD_TEST_MIN_LENGTH..PAYLOAD_TEST_MAX_LENGTH).each do |i|
+      # Compare this binary with the next one
+      diffs = diff(payload_array[i], payload_array[i+1])
+
+      break if diffs.nil?
+   
+      # Iterate through each diff, searching for offsets of the length and the payload
+      diffs.length.times do |j|
+        current_byte = diffs[j]
+        next_byte = diffs[j+1]
+        prev_byte = diffs[j-1]
+
+        if j > 0
+          # Skip this if we compared these two bytes on the previous iteration
+          next if prev_byte.position == current_byte.position
+        end
 
-      # Compare this byte and the following byte to identify length and buffer offsets
-      length_offsets.push(current_byte.position) if isLengthOffset?(current_byte,next_byte)
-      buffer_offsets.push(current_byte.position) if isBufferOffset?(current_byte,next_byte)
+        # Compare this byte and the following byte to identify length and buffer offsets
+        length_offsets.push(current_byte.position) if length_offset?(current_byte, next_byte)
+        buffer_offsets.push(current_byte.position - i) if buffer_offset?(current_byte, next_byte)
+      end
     end
-  end
 
-  if debug
-    for length_offset in length_offsets
-      STDERR.puts "  LENGTH OFFSET #{length_offset} = 0x#{empty_payload[length_offset-1].ord.to_s(16)} #{empty_payload[length_offset].ord.to_s(16)}"
+    if @debug
+      for length_offset in length_offsets
+        STDERR.puts "  LENGTH OFFSET #{length_offset} = 0x#{empty_payload[length_offset-1].ord.to_s(16)} #{empty_payload[length_offset].ord.to_s(16)}"
+      end
+
+      for buffer_offset in buffer_offsets
+        STDERR.puts "  BUFFER OFFSET #{buffer_offset}"
+      end
+      STDERR.puts "  PAYLOAD LENGTH: #{empty_payload.length}"
     end
 
-    for buffer_offset in buffer_offsets
-      STDERR.puts "  BUFFER OFFSET #{buffer_offset}"
+    payload_bytes = Base64.strict_encode64(empty_payload)
+    if buffer_offsets.length > 0
+      results[payload] = {
+        'status': 'dynamic',
+        'lengthOffset': length_offsets.uniq,
+        'bufferOffset': buffer_offsets.uniq,
+        'bytes': payload_bytes
+      }
+    else
+      #TODO: Turns out ysoserial doesn't have any static payloads.  Consider removing this.
+      results[payload] = {
+        'status': 'static',
+        'bytes': payload_bytes
+      }
     end
-    STDERR.puts "  PAYLOAD LENGTH: #{empty_payload.length}"
   end
+  results
+end
 
-  payloadBytes = Base64.strict_encode64(empty_payload).gsub(/\n/,"")
-  if buffer_offsets.length > 0
-    results[payload] = {
-      'status': 'dynamic',
-      'lengthOffset': length_offsets.uniq,
-      'bufferOffset': buffer_offsets.uniq,
-      'bytes': payloadBytes
-    }
+@payload_list = get_payload_list
+if @ysoserial_payloads 
+  unkown_list = @ysoserial_payloads - @payload_list
```

Comment:
Typo. This should probably be `unknown_list`. There's two other instances of `unkown_list`, below.

```suggestion
  unknown_list = @ysoserial_payloads - @payload_list
```